src/scene/materials/standard-material-options-builder.js

```diff
@@ -287,6 +288,10 @@ class StandardMaterialOptionsBuilder {
         options.litOptions.useIridescence = stdMat.useIridescence && stdMat.iridescence !== 0.0;
         options.litOptions.useMetalness = stdMat.useMetalness;
         options.litOptions.useDynamicRefraction = stdMat.useDynamicRefraction;
+
+        options.litOptions.useDisplacement = stdMat.useDisplacement;
```

Comment:
I suspect only useDisplacement needs to be in the options, as likely this is the only flag used by the generator. Not the values - those are just uniforms, and do not change the generated shader.

Reply:
`Lit-shader.js` uses this. Won't generate if does not set

`this.needsNormal ||= options.useDisplacement;` and       
 ```
if (options.useDisplacement) {
           code += "#define DISPLACEMENT\n";
}
```

Reply:
yes, that's what I said too. useDisplacement needs to be there, but not the other two values (following two lines)